nsdb-http/src/main/scala/io/radicalbit/nsdb/web/routes/DataApi.scala

```diff
@@ -75,21 +77,23 @@ trait DataApi {
     pathPrefix("data") {
       post {
         entity(as[InsertBody]) { insertBody =>
-          optionalHeaderValueByName(authenticationProvider.headerName) { header =>
-            authenticationProvider.authorizeMetric(ent = insertBody, header = header, writePermission = true) {
-              onComplete(
-                writeCoordinator ? MapInput(insertBody.bit.timestamp,
-                                            insertBody.db,
-                                            insertBody.namespace,
-                                            insertBody.metric,
-                                            insertBody.bit)) {
-                case Success(_: InputMapped) =>
-                  complete("OK")
-                case Success(RecordRejected(_, _, _, _, _, reasons, _)) =>
-                  complete(HttpResponse(BadRequest, entity = reasons.mkString(",")))
-                case Success(_) =>
-                  complete(HttpResponse(InternalServerError, entity = "unknown response"))
-                case Failure(ex) => complete(HttpResponse(InternalServerError, entity = ex.getMessage))
+          validateModel(insertBody).apply { validatedInsertBody =>
```

Comment:
I do not know if it's better to check security header before json body or viceversa. Any opinion is welcome

Reply:
IMO it's better to check security first because in the case you have not the right to access this API and you make an erroneous try you will get informatation about a field type and this is not safe.

Reply:
Agreed. In this case, it's better to put this line 

`validateModel(insertBody).apply { validatedInsertBody =>`

below 

`authenticationProvider.authorizeMetric`

Reply:
👍